cmd/tailscale/cli/set.go

```diff
@@ -61,6 +64,8 @@ func newSetFlagSet(goos string, setArgs *setArgsT) *flag.FlagSet {
 	setf.StringVar(&setArgs.hostname, "hostname", "", "hostname to use instead of the one provided by the OS")
 	setf.StringVar(&setArgs.advertiseRoutes, "advertise-routes", "", "routes to advertise to other nodes (comma-separated, e.g. \"10.0.0.0/8,192.168.0.0/24\") or empty string to not advertise routes")
 	setf.BoolVar(&setArgs.advertiseDefaultRoute, "advertise-exit-node", false, "offer to be an exit node for internet traffic for the tailnet")
+	setf.BoolVar(&setArgs.updateCheck, "update-check", true, "HIDDEN: automatically check for available Tailscale updates")
```

Comment:
do we still need this one if we can just push down this in MapResponse.ClientVersion? What's to "check"? Or is this really about whether the user wants to be notified about the availability of updates?

Reply:
Yeah, it's whether they will see GUI or CLI notifications. The assumption is that dinner users will want to run older versions and get annoyed at CLI messages telling them about updates

Reply:
Updated the flag description to better describe it